internal/services/compute/virtual_machine_scale_set_data_source.go

```diff
@@ -228,30 +222,38 @@ func getVirtualMachineScaleSetVMConnectionInfo(ctx context.Context, networkInter
 	privateIPAddresses := make([]string, 0)
 
 	for _, nic := range networkInterfaces {
-		for _, config := range *nic.IPConfigurations {
-			if props := config.InterfaceIPConfigurationPropertiesFormat; props != nil {
-				if pip := props.PublicIPAddress; pip != nil {
-					pipID, err := networkParse.VirtualMachineScaleSetPublicIPAddressID(*pip.ID)
-					if err != nil {
-						return nil, err
-					}
-
-					publicIPAddress, err := publicIPAddressesClient.GetVirtualMachineScaleSetPublicIPAddress(ctx, resourceGroupName, virtualMachineScaleSetName, virtualmachineIndex, pipID.NetworkInterfaceName, pipID.IpConfigurationName, pipID.PublicIPAddressName, "")
-					if err != nil {
-						return nil, fmt.Errorf("reading Public IP Address for VM Instance %q for Virtual Machine Scale Set %q (Resource Group %q): %+v", virtualmachineIndex, virtualMachineScaleSetName, resourceGroupName, err)
-					}
-
-					if pointer.From(nic.Primary) && pointer.From(props.Primary) {
-						primaryPublicAddress = *publicIPAddress.IPAddress
-					}
-					publicIPAddresses = append(publicIPAddresses, *publicIPAddress.IPAddress)
-				}
-
-				if props.PrivateIPAddress != nil {
-					if pointer.From(nic.Primary) && pointer.From(props.Primary) {
-						primaryPrivateAddress = *props.PrivateIPAddress
+		if props := nic.Properties; props != nil {
+			if ipConfigs := props.IPConfigurations; ipConfigs != nil {
+				for _, config := range *ipConfigs {
+					if configProps := config.Properties; configProps != nil {
+						if pip := configProps.PublicIPAddress; pip != nil {
+							pipID, err := commonids.ParseVirtualMachineScaleSetPublicIPAddressID(*pip.Id)
```

Comment:
This is coming from the API response so wants to be parsed insensitively:

```suggestion
							pipID, err := commonids.ParseVirtualMachineScaleSetPublicIPAddressIDInsensitively(*pip.Id)
```